compiler/codegen/symbol.cpp

```diff
@@ -2433,17 +2437,23 @@ void FnSymbol::codegenDef() {
         func->removeFnAttr(llvm::Attribute::ReadNone);
       }
 
+      // Skip the first LLVM formal if it is used for a struct return.
+      // Note that we only consume the first _LLVM_ formal. The Clang
+      // formal number remains 0.
```

Comment:
Did you check to see if the sret must be the first formal? I saw something hinting it might be the 2nd if the first formal is a `this`, somewhere. Anyway it seems like it'd be simple enough to move this check within the loop.

Reply:
Yes, there is a path for `returnInfo.isSretAfterThis()`, but it ends in an `INT_FATAL`. 

```cpp
1856       if (returnInfo.isSRetAfterThis()) {
1857         INT_FATAL("not handled"); // replace existing sret argument?
1858         clangArgNum++; // a guess
1859       }
```

Reply:
I can't even find documentation for `isSretAfterThis()`. Do we mark our receivers with the `this` attribute? Is LLVM even aware of Chapel receivers? If it is not it seems like it would not matter.

Reply:
In that case perhaps you could add an assert in the loop below checking for the sret attribute on `ai` and it can just assert that it isn't set?